src/Statement.php

```diff
@@ -82,7 +106,15 @@ public function execute($params = null)
      */
     public function bindValue($name, $value, $type = null)
     {
-        return $this->stmt->bindValue($name, $value, $type);
+        $args = func_get_args();
+
+        if (call_user_func_array([$this->stmt, __FUNCTION__], $args)) {
```

Comment:
would you use a direct call to `$this->stmt->bindValue()` instead ?
number and names of args are known, I see no need to use a less readable `call_user_func_array`.
I understand you may like to reuse `func_get_args` to both the stmt call and maintain the state but I would rather prefer to list the args (`$this->boundValues[$name] = [$name, $value, $type]`) for simplicity and readability.
afaik you did the same for `setFetchMode` just a little below..

Reply:
With `call_user_func_array` the default value for `$type` in the class of `$this->stmt` is used in case no argument for `$type` is given. The default was changed to `ParameterType::STRING` in doctrine/dbal: https://github.com/doctrine/dbal/blob/9efbf7f3e260442ceefe4aec163c18b193c8e0c2/lib/Doctrine/DBAL/Driver/Statement.php#L53. Though `null` is still accepted by dbal and the result will probably be the same, whether it is `null` or `ParameterType::STRING`... Nevertheless I agree with you that your suggestion is better readable.

Reply:
I see your point now!
what about fixing also the function declaration to match the Doctrine default then?

Reply:
Done.